hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/opt/OptUtils.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    /**
+     * Add a single trait to the trait set.
+     *
+     * @param traitSet Original trait set.
+     * @param trait Trait to add.
+     * @return Resulting trait set.
+     */
+    public static RelTraitSet traitPlus(RelTraitSet traitSet, RelTrait trait) {
+        return traitSet.plus(trait).simplify();
+    }
+
+    /**
+     * Add two traits to the trait set.
+     *
+     * @param traitSet Original trait set.
+     * @param trait1 Trait to add.
+     * @param trait2 Trait to add.
+     * @return Resulting trait set.
+     */
+    public static RelTraitSet traitPlus(RelTraitSet traitSet, RelTrait trait1, RelTrait trait2) {
+        return traitSet.plus(trait1).plus(trait2).simplify();
+    }
+
+    /**
+     * Convert the given trait set to logical convention.
+     *
+     * @param traitSet Original trait set.
+     * @return New trait set with logical convention.
+     */
+    public static RelTraitSet toLogicalConvention(RelTraitSet traitSet) {
+        return traitPlus(traitSet, HazelcastConventions.LOGICAL);
+    }
+
+    /**
+     * Convert the given input into logical input.
+     *
+     * @param input Original input.
+     * @return Logical input.
+     */
+    public static RelNode toLogicalInput(RelNode input) {
+        return convert(input, toLogicalConvention(input.getTraitSet()));
+    }
+
+    /**
+     * Convert the given trait set to physical convention.
+     *
+     * @param traitSet Original trait set.
+     * @return New trait set with physical convention and provided distribution.
+     */
+    public static RelTraitSet toPhysicalConvention(RelTraitSet traitSet) {
+        return traitPlus(traitSet, HazelcastConventions.PHYSICAL);
+    }
+
+    /**
+     * Convert the given trait set to physical convention.
+     *
+     * @param traitSet Original trait set.
+     * @param distribution Distribution.
+     * @return New trait set with physical convention and provided distribution.
+     */
+    public static RelTraitSet toPhysicalConvention(RelTraitSet traitSet, DistributionTrait distribution) {
+        return traitPlus(traitSet, HazelcastConventions.PHYSICAL, distribution);
+    }
+
+    /**
+     * Convert the given input into physical input.
+     *
+     * @param input Original input.
+     * @return Logical input.
+     */
+    public static RelNode toPhysicalInput(RelNode input) {
+        return convert(input, toPhysicalConvention(input.getTraitSet()));
+    }
+
+    /**
+     * Convert the given input into physical input.
+     *
+     * @param input Original input.
+     * @param distribution Distribution.
+     * @return Logical input.
+     */
+    public static RelNode toPhysicalInput(RelNode input, DistributionTrait distribution) {
+        return convert(input, toPhysicalConvention(input.getTraitSet(), distribution));
+    }
+
+    /**
+     * @param rel Node.
+     * @return {@code True} if the given node is physical node.
```

Comment:
Minor: True -> true?

Reply:
Too late, it is merged. I'll fix it in the next PR. Bad habit from Apache Ignite days - we had such word as a requirement to code styling :-)